pkg/controller/oneagent/oneagent_controller.go

```diff
@@ -363,18 +364,24 @@ func newPodSpecForCR(instance *dynatracev1alpha1.OneAgent) corev1.PodSpec {
 // Returns an error in the following conditions:
 //  - failure on object deletion
 //  - timeout on waiting for ready state
-func (r *ReconcileOneAgent) deletePods(instance *dynatracev1alpha1.OneAgent, pods []corev1.Pod) error {
+func (r *ReconcileOneAgent) deletePods(reqLogger logr.Logger, instance *dynatracev1alpha1.OneAgent, pods []corev1.Pod) error {
 	for _, pod := range pods {
+		reqLogger.Info("deleting pod", "pod", pod.Name)
+
 		// delete pod
 		err := r.client.Delete(context.TODO(), &pod)
 		if err != nil {
 			return err
 		}
 
+		reqLogger.Info("waiting until pod is ready", "pod", pod.Name)
+
 		// wait for pod on node to get "Running" again
 		if err := r.waitPodReadyState(instance, pod); err != nil {
 			return err
```

Comment:
Is it intentional, that an error for a `pod` happens, the entire function is exited?
Breaking the loop and continuing with the remaining ones could still be valid.

Reply:
Yeah, it makes sense to update what we can.